Ch7 issue16

Syakyr opened the pull request.

Commit:
Added #16 + extra requirements to render mermaid.js

Commit:
Changed a phrase to remove skunked word

chuawjk commented:
Hi @Syakyr, this is a good primer for engineers who are new to automation. A few suggestions for consideration:

1. Please rephrase the section title as a questions, since the handbook is intended to be in a Q&A style.

2. I think the `Start Small` subsection is really important. Perhaps you could consider adding a bit of content to communicate the need for iterative development. The iterative nature of building a CICD pipeline can also be added to diagram for more clarity.

3. Just an idea: do you think readers would benefit from the addition of a comparison table showing examples of manual vs. automated deployments? E.g., building Docker containers locally vs on the CICD pipeline; manually testing container pipelines vs doing so via CICD. I thought these would provide a more tangible context to readers as to the range of automation that CICD can help with.

Commit:
Added .swp files into gitignore

Commit:
Changed header + added reference in 'Start Small' subsection

Commit:
Removed extra 'the'

Syakyr commented:
Hi @chuawjk, hopefully the latest changes are fine:

> 1. Please rephrase the section title as a questions, since the handbook is intended to be in a Q&A style.

Done 👍🏼 

> 2. I think the `Start Small` subsection is really important. Perhaps you could consider adding a bit of content to communicate the need for iterative development. The iterative nature of building a CICD pipeline can also be added to diagram for more clarity.

I've added a reference point to another section written in issue #18 for a guide on where to start writing CI/CD for their sections.

> 3. Just an idea: do you think readers would benefit from the addition of a comparison table showing examples of manual vs. automated deployments? E.g., building Docker containers locally vs on the CICD pipeline; manually testing container pipelines vs doing so via CICD. I thought these would provide a more tangible context to readers as to the range of automation that CICD can help with.

Hmm, we could see whether there's a number of requests to add that comparison then we can do that then; for one, this section might be too long if I add that in, and that I've already added a situation of which they would find it useful if they're under that situation. I think the focus of the section is "if you're finding that doing manual testing is starting (or projecting) to get tedious, then this might be a guide to the solution for your issue", and if you're still fine with manual testing, then by all means continue to do so.

Commit:
Added contributor(s)

Commit:
Added link to CI/CD basics for reader's reference

Commit:
Updated overview for chapter 7

ngkh2018 commented:
1. I have no comments for the content, but a follow up question. Say, if the reader, after started small, wants to do more advanced CI/CD, do we have other guidelines/recommendations in the handbook to enable the reader, apart from the links included in the ‘References’ section?

Syakyr commented:
CI/CD as a whole is very platform-dependent at the moment, so there's no one pathway to recommend when it comes to advanced CI/CD. Github Actions is quite different from Gitlab CI, which is different from Jenkins. So around that time, they would need to look at the specific solutions they want to implement and go from there, which is against the handbook guidelines to be platform-agnostic as much as possible. 
But I suppose we could point to various CI/CD platforms for them to check out and see which ones stick for them, if that's suitable for the handbook itself.

najibninaba commented:
The edits look good to. Thanks @Syakyr!

kevinosl commented on book/7-solution-delivery/min-viable-code.md:
```diff
@@ -0,0 +1,100 @@
+
+## Start Small
+
+This final step in paramount as to not paralyse yourself in the vast 
```
```This final step in paramount as to not paralyse yourself in the vast ```
Is there a typo in this line?

Syakyr replied:
Yes, that's a typo; should be `is` instead of `in`. I have fixed it.

kevinosl commented:
Other than the above, looks good, no further comments.

Syakyr committed:
Fixed typo in line 71

ryzalk commented:
Closing this pull request in view of the new merging workflow i.e. from contributing branches to `develop`.
CC @chuawjk
